tests/bdp_spark_pi_submit_test.erl

```diff
@@ -0,0 +1,110 @@
+%will fail to link together.  As such, os:cmd is used to get the ip address of node1 via a shell command that might not work on systems other than moc osx.
+confirm() ->
+
+    %build cluster
+    lager:info("Building cluster"),
+    ClusterSize = 3,
+    _Nodes = [Node1, Node2, _Node3] =
+        bdp_util:build_cluster(
+          ClusterSize, [{lager, [{handlers, [{file, "console.log"}, {level, debug}] }]}]),
+   
+    %% add spark master and worker services
+    lager:info("Adding Spaker Master and Spark Worker Service..."), 
+    %% add master service
+    ok = bdp_util:add_service(Node1, ?SPARK_MASTER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE, ?SPARK_MASTER_SERVICE_CONFIG),
+    ok = bdp_util:wait_services(Node1, {[], [?SPARK_MASTER_SERVICE_NAME]}),
+    %% add worker service
+    SparkMasterIP = re:replace(rpc:call(Node1,os,cmd,["ipconfig getifaddr en0"]),"\\s+", "", [global,{return,list}]),
+    SparkMasterAddr = "spark://"++SparkMasterIP++":7077",
+    WorkerConfig = [{"MASTER_URL", SparkMasterAddr},{"SPARK_WORKER_PORT","8081"}],
+
+    ok = bdp_util:add_service(Node2, ?SPARK_WORKER_SERVICE_NAME, ?SPARK_WORKER_SERVICE_TYPE, WorkerConfig),
+    ok = bdp_util:wait_services(Node2, {[], [?SPARK_MASTER_SERVICE_NAME, ?SPARK_WORKER_SERVICE_NAME]}),
+
+    lager:info("Service ~p (~s) added", [?SPARK_MASTER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE]),
+    %start master and worker
+    ok = bdp_util:start_service(Node1, Node1, ?SPARK_MASTER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE),
+    ok = bdp_util:wait_services(Node1, {[?SPARK_MASTER_SERVICE_NAME], [?SPARK_MASTER_SERVICE_NAME, ?SPARK_WORKER_SERVICE_NAME]}),
+    ok = bdp_util:start_service(Node2, Node2, ?SPARK_WORKER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE),
+    ok = bdp_util:wait_services(Node2, {[?SPARK_MASTER_SERVICE_NAME,?SPARK_WORKER_SERVICE_NAME], [?SPARK_MASTER_SERVICE_NAME, ?SPARK_WORKER_SERVICE_NAME]}),
+
+    lager:info("Waiting 10 seconds..."),
+    timer:sleep(10000),
+
+    %check running services
+    {Run,Ava} = bdp_util:get_services(Node1),
+
+    MyServices = ["my-spark-master","my-spark-worker"],
+    ?assert(Run == MyServices),
+    ?assert(Ava == MyServices),
+    lager:info("~s",[MyServices]),
+    lager:info("Running: ~s", [Run]),
+    lager:info("Available: ~s", [Ava]),
+
+    %%run spark job pi.py
+    SparkJobSubmit1 = "./lib/data_platform-1/priv/spark-master/bin/spark-submit --master ",
+    SparkJobSubmit2 = " ./lib/data_platform-1/priv/spark-master/examples/src/main/python/pi.py 100",
+    SparkJobSubmit3 = SparkJobSubmit1++SparkMasterAddr++SparkJobSubmit2,
+    Results = rpc:call(Node2,os,cmd,[SparkJobSubmit3]),
+    lager:info("Spark Job Results: ~s", [Results]),
+
+    %Test if the spark job submission worked
+    ?assert(string:str(Results,"Pi is roughly") > 0),
+
+    %Assert proper cluster execution of spark job
+    WgetCall = "wget "++SparkMasterIP++":8080",
+    lager:info("wget call: ~s",[WgetCall]),
+    rpc:call(Node1,os,cmd,[WgetCall]),
```

Comment:
Just a note: you don't have to do an RPC call here. Just run it directly on riak_test node.